### DISCUSSION TITLE
How do I disable digest pinning in docker-compose, but keep it in dockerfiles?

### DISCUSSION THREAD

**Discussion by annervisser** [renovatebot/renovate#22974] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want renovate to create a pinDigest pull request for just our dockerfiles, not our docker-compose. I DO want it to bump versions in docker-compose.

If at all possible, it would be nice if renovate could update any dependencies in docker-compose that are already pinned to a digest.

### What you tried so far.

My full config: 
```json5
{
	"$schema": "https://docs.renovatebot.com/renovate-schema.json",
	"extends": [
		"config:base",
		":automergeLinters",
		":automergeRequireAllStatusChecks",
		":automergeTypes",
		":combinePatchMinorReleases",
		":renovatePrefix",
		":semanticCommitsDisabled",
		"docker:pinDigests",
	],
	"schedule": [
		"after 6pm every weekday",
		"before 7am every weekday",
		"every weekend",
	],
	"enabledManagers": [
		"dockerfile",
		"docker-compose",
		"github-actions",
	],
	"docker-compose": {
		"digest": {
			"enabled": false,
		},
	},
}

```

I've tried the docker-compose config option:
```json5
	"docker-compose": {
		"digest": {
			"enabled": false,
		},
	},
```
Then a packagerule about pinning:
```json5
	"packageRules": [
		// Don't auto-pin dependencies in docker-compose, pinned dependencies will still be updated
		{
			"managers": [
				"docker-compose"
			],
			"updateTypes": [
				"pin",
			],
			"enabled": false,
		},
	],
```

Then also disabling digests:
```json5
	"packageRules": [
		// Don't auto-pin dependencies in docker-compose, pinned dependencies will still be updated
		{
			"managers": [
				"docker-compose"
			],
			"updateTypes": [
				"pin",
				"digest",
			],
			"enabled": false,
		},
	],
```

All of these resulted in a pull request where both dockerfiles and docker-compose are pinned to a digest.

**Comment by rarkins** [comment, marked as answer]
Remove `docker:pinDigests` and instead add a `packageRules` object with `matchManagers=dockerfile` and `pinDigests=true`
